### ISSUE TITLE
Thread-safe

### ISSUE THREAD

**Issue by mehmetilker** [zzzprojects/EntityFramework-Plus#32]
I could not see any explanation if caching extension is thread safe.
Could you give some information if appropriate lock strategy is applied. Or do we need to take care pf that part ?

**Comment by zzzprojects** [comment]
Yes, EF+ Query Cache is thread safe. We use a memory cache and concurrent dictionary to ensure to be thread safe.

The default value is InfiniteAbsoluteExpiration (never expire): https://msdn.microsoft.com/en-us/library/system.runtime.caching.cacheitempolicy.absoluteexpiration(v=vs.110).aspx

Let me know if that answer to your questions.

**Comment by mehmetilker** [comment]
Thank you for your answer and effort on the project.

I would like to add one recommendation as a feature as I see you are developing extensions for EF.

It would be nice to have an extension (possible middleware) allow us to specify lookup tables to be cached in startup.

**Comment by zzzprojects** [comment]
I'm not sure to understand the feature you are requesting, can you explain it?

**Comment by mehmetilker** [comment]
Let's say I have a ProductCode table with 5000 row in it. Doesn't change often, whole table can be cached, and takes 3 seconds to load. So when a product details page viewed it will wait for 3 second to load ProductCode table to the cache.

I can make a call to repository layer to trigger loading table to the cache from application startup which is a dirty solution.

Or I can specify which tables should be cached while application is warming up.
Something like how we tell EF should read connection string.
```

services.AddDbContext<ApplicationDbContext>(
options =>
                options.UseSqlServer(Configuration.GetConnectionString("DbConnection")));
```


An example could be:

```

services.CacheForDbContext<ApplicationDbContext>(options =>

                options.CacheTable<ProductCode>();

                options.CacheTable<ServiceCode>();

                options.CacheTable<PostalCode>();

));

```

**Comment by zzzprojects** [comment]
Thank you,

I will look at this suggestion very soon and propose you a solution.

Best Regards,

Jonathan
